src/database/remove-remote.ts

```diff
@@ -0,0 +1,126 @@
+"use strict";
+
+import { Response } from "request";
+import * as request from "request";
+import responseToError = require("../responseToError");
+import utils = require("../utils");
+import FirebaseError = require("../error");
+import logger = require("../logger");
+import api = require("../api");
+
+class RemoveRemote {
+  private instance: string;
+
+  constructor(instance: string) {
+    this.instance = instance;
+  }
+
+  public deletePath(path: string): Promise<boolean> {
+    return new Promise((resolve, reject) => {
+      const url =
+        utils.addSubdomain(api.realtimeOrigin, this.instance) + path + ".json?print=silent";
+      const reqOptions = {
+        url,
+        json: true,
+      };
+      return api.addRequestHeaders(reqOptions).then((reqOptionsWithToken) => {
+        request.del(reqOptionsWithToken, (err: any, res: Response, body: any) => {
+          if (err) {
+            return reject(
+              new FirebaseError("Unexpected error while removing data at " + path, {
+                exit: 2,
+                original: err,
+              })
+            );
+          } else if (res.statusCode >= 400) {
+            return reject(responseToError(res, body));
+          }
+          logger.debug("[database] Sucessfully removed data at " + path);
+          return resolve(true);
+        });
+      });
+    });
+  }
+
+  public prefetchTest(path: string): Promise<string> {
+    const url =
+      utils.addSubdomain(api.realtimeOrigin, this.instance) + path + ".json?timeout=100ms";
+    const reqOptions = {
+      url,
+    };
+    return api.addRequestHeaders(reqOptions).then((reqOptionsWithToken) => {
+      return new Promise<string>((resolve, reject) => {
+        logger.debug("[database] Prefetching test at " + path);
+        request.get(reqOptionsWithToken, (err: any, res: Response, body: any) => {
+          if (err) {
+            return reject(
+              new FirebaseError("Unexpected error while prefetching data to delete" + path, {
+                exit: 2,
+              })
+            );
+          }
+          switch (res.statusCode) {
+            case 200:
+              if (body) {
+                return resolve("small");
```

Comment:
I would highly recommend defining an `enum` with these return values (empty, small, large) and using that type in the code. It'll help catch mistakes if a correct type isn't returned.

See "String enums" on this page: https://www.typescriptlang.org/docs/handbook/enums.html